File: middlerApp.API/IDP/DtoModels/MUserListDto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using middlerApp.API.IDP.Models;
using Org.BouncyCastle.Asn1;

namespace middlerApp.API.IDP.DtoModels
{
    public class MUserListDto
    {

        public Guid Id { get; set; }

        public string UserName { get; set; }

        public string Email { get; set; }
        public bool EmailConfirmed { get; set; }

        public string FirstName { get; set; }
        public string LastName { get; set; }

        public string PhoneNumber { get; set; }
        public bool PhoneNumberConfirmed { get; set; }

        public bool HasPassword { get; set; }

        public bool Active { get; set; }

        public ICollection<string> Logins { get; set; } = new List<string>();

    }
}

File: middlerApp.UI/projects/AdminUI/src/app/components/idp/models/m-user-list-dto.ts
export class MUserListDto {
    Id: string;
    UserName: string;
    Email: string;
    Firstname: string;
    Lastname: string;

    EmailConfirmed: boolean;
    Active: boolean;
    
    HasPassword: boolean;
    Logins: Array<string>;
    Deleted: boolean;
}



export function MUserListDtoSortByName(a: MUserListDto, b: MUserListDto) {
    if (a.UserName?.toLowerCase() < b.UserName?.toLowerCase()) {
        return -1;
    }
    if (a.UserName?.toLowerCase() > b.UserName?.toLowerCase()) {
        return 1;
    }
    return 0;
}